hermes/file_gen.go

```diff
@@ -1,28 +1,28 @@
-package main
+package hermes
+
 import (
 	"fmt"
-	// "math/rand"
 	"strconv"
 	"crypto/sha1"
 )
 type HermesFile struct {
 	name string
 	contents string
 }
-func generate_file_name (file_ID int, file_checksum string) string{
-	var id_string string=""
+func (file *HermesFile) GenerateFileName(file_ID int, file_checksum string) {
```

Comment:
- I don't think that the file name generating function should be made public.

- nit: if you use snake_case, all your alphabetic characters should be in lower case. Ditto for all other similar situations in the rest of the file.

```suggestion

func (file *HermesFile) generateFileName(file_id int, file_checksum string) {
```